src/System.Net.Security/src/System/Net/Security/SslSyncAsyncAdapters.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Security
+{
+    internal interface ISslWriteAdapter
+    {
+        Task LockAsync();
+        void Release();
```

Comment:
Not sure having Release on the interface is useful.  It's not different b/w sync and async.

Reply:
That is true. The only reason i did it was because i am not 100% sure if it should be sync or not. It currently boots any pending handshake if you are writing off to another thread.

Interestingly the previous implementation has a comment also asking if that is the right thing to do ....

Reply:
It does feel a bit awkward there but it is also the only place the write side needs the sslstream ref and it kinda goes with the lock unlock logic... but I am on the fence so if you think it should go I am cool with that, your choice

Reply:
Removed the Release, it's not just release anyway its a finishwrite so makes sense to be removed